src/pages/instances/InstanceDetailHeaderDropdown.tsx

```diff
@@ -0,0 +1,58 @@
+import { FC } from "react";
+import DeleteInstanceBtn from "./actions/DeleteInstanceBtn";
+import { LxdInstance } from "types/instance";
+import MigrateInstanceBtn from "./actions/MigrateInstanceBtn";
+import { useSettings } from "context/useSettings";
+import { isClusteredServer } from "util/settings";
+import CreateImageFromInstanceBtn from "./actions/CreateImageFromInstanceBtn";
+import DuplicateInstanceBtn from "./actions/DuplicateInstanceBtn";
+import { ContextualMenu } from "@canonical/react-components";
+
+interface Props {
+  instance?: LxdInstance;
+  project: string;
+  isLoading: boolean;
+}
+
+const InstanceDetailHeaderDropdown: FC<Props> = ({
+  instance,
+  project,
+  isLoading,
+}) => {
+  const { data: settings } = useSettings();
+  const isClustered = isClusteredServer(settings);
+
+  return (
+    <>
+      {instance ? (
+        <ContextualMenu toggleLabel="Options" hasToggleIcon visible>
+          {isClustered ? (
+            <MigrateInstanceBtn
+              instance={instance}
+              project={project}
+              dropdownItemClassname="dropdownItemClassname"
+            />
+          ) : null}
+          <CreateImageFromInstanceBtn
+            key="publish"
+            instance={instance}
+            dropdownItemClassname="dropdownItemClassname"
+          />
```

Comment:
This should look like in the [storybook example](https://canonical.github.io/react-components/?path=/story/components-contextualmenu--toggle). Without border and left aligned text. We might be able to pass the `p-contextual-menu__link` class to the buttons to get it get the layout correct. The custom `dropdownItemClassname` is probably not needed then.